core/src/main/java/com/avast/bytes/AbstractBytesComparator.java

```diff
@@ -0,0 +1,35 @@
+package com.avast.bytes;
+
+
+import java.util.Comparator;
+
+public abstract class AbstractBytesComparator
+        implements Comparator<Bytes> {
+
+    private final boolean shorterFirst;
+
+    protected AbstractBytesComparator() {
+        this(true);
+    }
+
+    /**
+     * @param shorterFirst If true, shorter Bytes with same prefix will be take as less
+     *                     For example, 0x01 < 0x0102
+     */
+    protected AbstractBytesComparator(boolean shorterFirst) {
+        this.shorterFirst = shorterFirst;
+    }
+
+    @Override
+    public int compare(Bytes a, Bytes b) {
+        int n = Math.min(a.size(), b.size());
+        for (int position = 0; position < n; position++) {
+            int cmp = compare(a.byteAt(position), b.byteAt(position));
+            if (cmp != 0)
```

Comment:
I would like to see curly braces around this. I think it can prevent some problems when you edit the code in the future.